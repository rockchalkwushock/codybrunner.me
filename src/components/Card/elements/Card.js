import styled from 'styled-components'

const Card = styled.div`
  padding: 0.5em;
  margin: 0 auto;
  @media (min-width: ${({ theme }) => theme.screen.large}) {
    width: 60%;
  }
`

export default Card
