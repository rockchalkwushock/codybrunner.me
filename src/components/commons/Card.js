import React from 'react'
import styled from 'styled-components'

import Link from './Link'

const CardContainer = styled.div`
  border: 1px solid
    ${({ alt, theme }) =>
      alt === 'true' ? theme.colors.secondary : theme.colors.alt};
  padding: 1rem;
  margin: 0.75rem auto;
  max-width: ${({ theme }) => theme.maxWidth};
`

const CardTitle = styled.h1`
  color: ${({ alt, theme }) =>
    alt === 'true' ? theme.colors.secondary : theme.colors.alt};
  font-size: 1.55rem;
  margin: unset;
  text-align: center;

  @media (min-width: ${({ theme }) => theme.screen.large}) {
    font-size: 2rem;
  }
`

const CardDate = styled.h5`
  color: ${({ alt, theme }) =>
    alt === 'true' ? theme.colors.secondary : theme.colors.alt};
  margin: unset;
  padding-top: 0.5rem;
  text-align: center;
`

const CardInfo = styled.p`
  color: ${({ alt, theme }) =>
    alt === 'true' ? theme.colors.secondary : theme.colors.alt};
  margin: unset;
  padding-top: 0.5rem;
  text-align: center;
`

const CardLinks = styled.div`
  display: flex;
  justify-content: space-evenly;
  padding-top: 0.5rem;
`

export default ({ alt, date, href, info, src, title }) => (
  <CardContainer alt={alt}>
    <CardTitle alt={alt}>{title}</CardTitle>
    {href && src ? null : <CardDate alt={alt}>{date}</CardDate>}
    <CardInfo alt={alt}>{info}</CardInfo>
    {href && src ? (
      <CardLinks>
        <Link altClr ext href={src}>
          Source
        </Link>
        <Link altClr ext href={href}>
          Live
        </Link>
      </CardLinks>
    ) : null}
  </CardContainer>
)