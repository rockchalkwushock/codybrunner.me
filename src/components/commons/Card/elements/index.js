import styled from 'styled-components'

const CardContainer = styled.div`
  border: 1px solid black;
  padding: 1rem;
  margin: 0.75rem 0;
`

const CardTitle = styled.h1`
  margin: unset;
`

const CardDate = styled.h5`
  margin: unset;
  padding-top: 0.5rem;
`

const CardInfo = styled.p`
  margin: unset;
  padding-top: 1rem;
`

export { CardContainer, CardDate, CardInfo, CardTitle }
