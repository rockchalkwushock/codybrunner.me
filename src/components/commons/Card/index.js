import React from 'react'

import Link from '../Link'
import {
  CardContainer,
  CardDate,
  CardInfo,
  CardLinks,
  CardTitle
} from './elements'

const Card = ({ alt, date, href, info, src, title }) => (
  <CardContainer alt={alt}>
    <CardTitle alt={alt}>{title}</CardTitle>
    {href && src ? null : <CardDate alt={alt}>{date}</CardDate>}
    <CardInfo alt={alt}>{info}</CardInfo>
    {href && src ? (
      <CardLinks>
        <Link alt ext href={src}>
          Source
        </Link>
        <Link alt ext href={href}>
          Live
        </Link>
      </CardLinks>
    ) : null}
  </CardContainer>
)

export default Card
