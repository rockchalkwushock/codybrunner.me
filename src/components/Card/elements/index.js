import Card from './Card'
import CardDate from './Date'
import CardExcerpt from './Excerpt'
import CardTags, { Tag } from './Tags.js'
import CardTitle from './Title'

export { Card, CardDate, CardExcerpt, CardTags, CardTitle, Tag }
