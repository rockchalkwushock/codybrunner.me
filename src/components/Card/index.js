import React from 'react'
import { arrayOf, bool, shape, string } from 'prop-types'

import { Link, Tags } from '../commons'
import { Card, CardDate, CardExcerpt, CardTitle } from './elements'

/**
 * @fileOverview
 * Card Component
 *
 * Accepts individual post data from GraphQL Query.
 */

const SiteCard = ({ post }) => (
  <Card draft={post.frontmatter.draft}>
    <CardTitle>
      <Link href={post.fields.slug} text={post.frontmatter.title} />
    </CardTitle>
    <CardDate>{post.frontmatter.date}</CardDate>
    <CardExcerpt>{post.excerpt}</CardExcerpt>
    <Tags tags={post.frontmatter.tags} />
  </Card>
)

SiteCard.propTypes = {
  post: shape({
    excerpt: string.isRequired,
    fields: shape({
      slug: string.isRequired
    }),
    frontmatter: shape({
      date: string.isRequired,
      draft: bool.isRequired,
      tags: arrayOf(string).isRequired,
      title: string.isRequired
    })
  })
}

export default SiteCard
