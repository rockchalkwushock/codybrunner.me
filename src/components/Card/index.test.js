import React from 'react'
import { shallow } from 'enzyme'

import Card from './'

const post = {
  excerpt: 'blah blah',
  fields: {
    slug: '/turdsspace/'
  },
  frontmatter: {
    date: '02-03-99',
    draft: true,
    tags: ['one', 'two'],
    title: 'All About JS'
  }
}

describe('Component: <Card />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Card post={post} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
