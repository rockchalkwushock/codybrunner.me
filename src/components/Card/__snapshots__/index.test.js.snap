// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Card /> should render correctly 1`] = `
<Card
  draft={true}
>
  <Title>
    <SiteLink
      href="/turdsspace/"
      text="All About JS"
    />
  </Title>
  <Date>
    02-03-99
  </Date>
  <Excerpt>
    blah blah
  </Excerpt>
  <Tags
    tags={
      Array [
        "one",
        "two",
      ]
    }
  />
</Card>
`;
